File: frontend/src/constants/activity.ts
// 活动状态与类型枚举（与后端 backend/internal/constants/activity.go 保持一致）
export const ActivityStatus = {
  DRAFT: 'draft',
  PUBLISHED: 'published',
  ENDED: 'ended',
} as const

export type ActivityStatusType = (typeof ActivityStatus)[keyof typeof ActivityStatus]

export const ActivityStatusText: Record<string, string> = {
  [ActivityStatus.DRAFT]: '草稿',
  [ActivityStatus.PUBLISHED]: '已发布',
  [ActivityStatus.ENDED]: '已结束',
}

export const ActivityType = {
  LECTURE: 'lecture',
  TRAINING: 'training',
  PARTY: 'party',
  COMPETITION: 'competition',
} as const

export type ActivityTypeType = (typeof ActivityType)[keyof typeof ActivityType]

export const ActivityTypeText: Record<string, string> = {
  [ActivityType.LECTURE]: '讲座',
  [ActivityType.TRAINING]: '培训',
  [ActivityType.PARTY]: '聚会',
  [ActivityType.COMPETITION]: '比赛',
}

export const ActivityTypeOptions = Object.values(ActivityType).map((v) => ({
  label: ActivityTypeText[v],
  value: v,
}))

export const ActivityStatusOptions = Object.values(ActivityStatus).map((v) => ({
  label: ActivityStatusText[v],
  value: v,
}))

File: backend/internal/constants/activity.go
package constants

// ActivityStatus 活动状态枚举（前后端重复定义，新增值需同步修改 ≥10 处文件）。
const (
	ActivityStatusDraft     = "draft"
	ActivityStatusPublished = "published"
	ActivityStatusEnded     = "ended"
)

// ActivityStatusValues 全部活动状态值。
var ActivityStatusValues = []string{ActivityStatusDraft, ActivityStatusPublished, ActivityStatusEnded}

// ActivityType 活动类型枚举。
const (
	ActivityTypeLecture     = "lecture"
	ActivityTypeTraining    = "training"
	ActivityTypeParty       = "party"
	ActivityTypeCompetition = "competition"
)

// ActivityTypeValues 全部活动类型值。
var ActivityTypeValues = []string{ActivityTypeLecture, ActivityTypeTraining, ActivityTypeParty, ActivityTypeCompetition}

// IsValidActivityStatus 校验活动状态。
func IsValidActivityStatus(s string) bool {
	for _, v := range ActivityStatusValues {
		if v == s {
			return true
		}
	}
	return false
}

// IsValidActivityType 校验活动类型。
func IsValidActivityType(s string) bool {
	for _, v := range ActivityTypeValues {
		if v == s {
			return true
		}
	}
	return false
}
